citation-matching/citation-matching-core-code/src/main/java/pl/edu/icm/coansys/citations/CitationMatchingJob.java

```diff
@@ -80,60 +53,69 @@ public static void main(String[] args) throws IOException, InstantiationExceptio
 
 
 
-
     //------------------------ PRIVATE --------------------------
     
-    private static JavaPairRDD<String, MatchableEntity> loadEntities(JavaSparkContext sc, String entitiesFilePath, Integer numberOfPartitions) {
+    @SuppressWarnings("rawtypes")
+    private static ConfigurableCitationMatchingService createConfigurableCitationMatchingService(JavaSparkContext sc, CitationMatchingJobParameters params) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
         
-        JavaPairRDD<Text, BytesWritable> readEntities = null; 
+        ConfigurableCitationMatchingService configurableCitationMatchingService = new ConfigurableCitationMatchingService();
         
-        if (numberOfPartitions == null) {
-            readEntities = sc.sequenceFile(entitiesFilePath, Text.class, BytesWritable.class);
-        } else {
-            readEntities = sc.sequenceFile(entitiesFilePath, Text.class, BytesWritable.class, numberOfPartitions);
-        }
+        CoreCitationMatchingService coreCitationMatchingService = createCoreCitationMatchingService(sc, params);
+        
+        configurableCitationMatchingService.setCoreCitationMatchingService(coreCitationMatchingService);
+        configurableCitationMatchingService.setNumberOfPartitions(params.numberOfPartitions);
         
-        JavaPairRDD<String, MatchableEntity> entities = readEntities.mapToPair(x -> new Tuple2<String, MatchableEntity>(x._1.toString(), MatchableEntity.fromBytes(x._2.copyBytes())));
+        createAndSetInputCitationReaderAndConverter(configurableCitationMatchingService, sc, params);
+        createAndSetInputDocumentReaderAndConverter(configurableCitationMatchingService, sc, params);
+        createAndSetOutputWriterAndConverter(configurableCitationMatchingService, sc, params);
         
-        numberOfPartitions = readEntities.partitions().size();
+        return configurableCitationMatchingService;
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private static void createAndSetInputCitationReaderAndConverter(ConfigurableCitationMatchingService citationMatchingService, JavaSparkContext sc, CitationMatchingJobParameters params) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
```

Comment:
You should not overuse warning suppression. In this case you could write e.g.
`    //@SuppressWarnings("rawtypes")
    private static ConfigurableCitationMatchingService<?,?,?,?,?,?> createConfigurableCitationMatchingService(JavaSparkContext sc, CitationMatchingJobParameters params) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
        
        ConfigurableCitationMatchingService<?,?,?,?,?,?> configurableCitationMatchingService = new ConfigurableCitationMatchingService<>();
        
        CoreCitationMatchingService coreCitationMatchingService = createCoreCitationMatchingService(sc, params);
        
        configurableCitationMatchingService.setCoreCitationMatchingService(coreCitationMatchingService);
        configurableCitationMatchingService.setNumberOfPartitions(params.numberOfPartitions);
        
        createAndSetInputCitationReaderAndConverter(configurableCitationMatchingService, sc, params);
        createAndSetInputDocumentReaderAndConverter(configurableCitationMatchingService, sc, params);
        createAndSetOutputWriterAndConverter(configurableCitationMatchingService, sc, params);
        
        return configurableCitationMatchingService;
    }

    //@SuppressWarnings({ "rawtypes", "unchecked" })
    private static <ICK, ICV, IDK, IDV, OMK, OMV> void createAndSetInputCitationReaderAndConverter(ConfigurableCitationMatchingService<ICK, ICV, IDK, IDV, OMK, OMV> citationMatchingService, JavaSparkContext sc, CitationMatchingJobParameters params) throws InstantiationException, IllegalAccessException, ClassNotFoundException {
        
        InputCitationReader<ICK, ICV> inputCitationReader = Class.forName(params.inputCitationReaderClass).asSubclass(InputCitationReader.class).newInstance();
        inputCitationReader.setSparkContext(sc);
        citationMatchingService.setInputCitationReader(inputCitationReader);
        
        InputCitationConverter<ICK, ICV> inputCitationConverter = Class.forName(params.inputCitationConverterClass).asSubclass(InputCitationConverter.class).newInstance();
        citationMatchingService.setInputCitationConverter(inputCitationConverter);
        
    }
`